File: lua/7-logika.lua
#!/bin/lua

function iloczyn(a, b)
	print("iloczyn", a, "and", b, "=", a and b)
end

function suma(a, b)
	print("suma", a, "or", b, "=", a or b)
end

function suma_xor(a, b)
	print("suma modulo2", a, "~", b, "=", a ~ b)
end

function negacja( a)
	print("negacja not " .. a, "=", not a)
end

zero=0
jeden=1

iloczyn(zero, zero)
iloczyn(zero, jeden)
iloczyn(jeden, zero)
iloczyn(jeden, jeden)

suma(zero, zero)
suma(zero, jeden)
suma(jeden, zero)
suma(jeden, jeden)

suma_xor(zero, zero)
suma_xor(zero, jeden)
suma_xor(jeden, zero)
suma_xor(jeden, jeden)

negacja(zero)
negacja(jeden)

iloczyn(false, false)
iloczyn(false, true)
iloczyn(true, false)
iloczyn(true, true)

suma(false, false)
suma(false, true)
suma(true, false)
suma(true, true)

if (true)
then
	print("true")
end	

if (not true)
then
	print("true")
else	
	print("not true")
end	

a=3
b=1
c=2
-- od lua 5.3
print("iloczyn:", a, "&", b, "=", a & b)
print("suma:", b, "|", c, "=", b | c)
print("suma modulo 2 (xor)", b, "~", c, "=", b ~ c)
print("negacja ~", b, "=", ~b)

File: objc/7-logika.m
#include <stdio.h>

void iloczyn(char a, char b)
{
	printf("iloczyn %d && %d = %d\n", a, b, a && b);
}

void suma(char a, char b)
{
	printf("suma %d || %d = %d\n", a, b, a || b);
}

void suma_xor(char a, char b)
{
	printf("suma xor %d ^ %d = %d\n", a, b, a ^ b);
}

void negacja(char a)
{
	printf("negacja !%d  = %d\n", a, !a);
}


int main()
{
	char zero=0, jeden=1;

	iloczyn(zero, zero);
	iloczyn(zero, jeden);
	iloczyn(jeden, zero);
	iloczyn(jeden, jeden);

	suma(zero, zero);
	suma(zero, jeden);
	suma(jeden, zero);
	suma(jeden, jeden);
	
	suma_xor(zero, zero);
	suma_xor(zero, jeden);
	suma_xor(jeden, zero);
	suma_xor(jeden, jeden);

	negacja(zero);
	negacja(jeden);

	char a=3;
	char b=1;
	char c=2;
	
	printf("iloczyn: %d & %d = %d\n", a, b, a & b);
	printf("suma: %d | %d = %d\n", b, c, b & c);
	printf("suma modulo 2 (xor) %d ^ %d = %d\n", b, c, b ^ c);
	printf("negacja ~%d = %d\n", b, ~b);


	return 0;
}
